schemas/macro_service.schema.json

```diff
@@ -0,0 +1,52 @@
+{
+  "$id": "https://github.com/dart-lang/macros/blob/main/schemas/macro_service.schema.json",
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "oneOf": [
+    {"$ref": "#/$defs/AugmentRequest"},
+    {"$ref": "#/$defs/AugmentResponse"}
+  ],
+  "$defs": {
+    "AugmentRequest": {
+      "type": "object",
+      "description": "A request to a macro to augment some code.",
+      "properties": {
+        "phase": {
+          "description": "Which phase to run: 1, 2 or 3.",
+          "type": "integer"
+        }
+      }
+    },
+    "AugmentResponse": {
+      "type": "object",
+      "description": "Macro's response to an [AugmentRequest]: the resulting augmentations."
+    },
+    "HostEndpoint": {
+      "type": "object",
+      "description": "A macro host server endpoint. TODO(davidmorgan): this should be a oneOf supporting different types of connection. TODO(davidmorgan): it's not clear if this belongs in this package! But, where else?",
+      "properties": {
+        "port": {
+          "description": "TCP port to connect to.",
+          "type": "integer"
+        }
+      }
+    },
+    "MacroDescription": {
+      "type": "object",
+      "description": "Information about a macro that the macro provides to the host.",
+      "properties": {
+        "runsInPhaseOne": {
```

Comment:
I would consider a list of phases as opposed to 3 bools, could just be a list of integers.